### ISSUE TITLE
Auto-cleanup mode

### ISSUE THREAD

**Issue by cpcallen** [rachel-fenichel/blockly-keyboard-experimentation#119]
In discussing various possible block-dragging scenarios , it has several times come up that it might be helpful—particularly to visually-impaired users who depend on a screen reader—if Blockly had a mode where the top-level blocks on the workspace were always arranged into single vertical column.

There is already a method, `WorkspaceSvg.prototype.cleanUp` that is called from the cleanup shortcut that can probably be used to handle most of the work.  There are a few details that should probably be considered, however:

- Care will need to be taken in dealing with block dragging (either by mouse or by keyboard nav)—e.g. it should be possible to position a new top level block (whether created from the toolbox or detached from another stack) above, below or between any existing top-level stacks, with the other stacks being repositioned accordingly—but does the "cleanup" happen as the block is being dragged about (e.g. as iOS homescreenp icons behave when being rearranged), or only after it is dropped?
- Care should also be taken, that when loading programs that already have a 2d block layout they are not rearranged into a single column without the user's explicit consent.
  - Relatedly: we should verify that the implementation preserves code generation order—i.e., that loading, cleaning and saving a program cannot result in any change to the generated code for the program.  (It appears that the existing `cleanUp` does preserve order.)
- This feature should probably ultimately live either in core or in a separate plugin, though it might be useful to implement it here first as it would probably be useful when UX testing keyboard nav + screen reader support.
